oauth2client/contrib/django_util/__init__.py

```diff
@@ -299,7 +403,11 @@ def has_credentials(self):
     @property
     def credentials(self):
         """Gets the authorized credentials for this flow, if they exist"""
-        return storage.get_storage(self.request).get()
+        if oauth2_settings.storage_model:
+            if self.request.user.is_authenticated():
+                return get_storage(self.request).get()
```

Comment:
the `else` case here ends up returning `None` but it's not very explicit. It might make sense to explicitly spell it out.

Also it seems that there's only one case where you don't do `get_storage(self.request).get()`, it might make sense to re-organize this?